src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/event/AsyncGoalEventService.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.event
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import mu.KotlinLogging
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.TelemetryService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.GoalEventService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEvent.Companion.newTimelineEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEventType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.service.TimelineService
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Implementation of [GoalEventService] for performing additional asynchronous actions related to [Goal] events.
+ */
+@Component
+class AsyncGoalEventService(
+  private val telemetryService: TelemetryService,
+  private val timelineEventResolver: TimelineEventResolver,
+  private val timelineService: TimelineService,
+) : GoalEventService {
+
+  override fun goalCreated(prisonNumber: String, createdGoal: Goal) {
+    runBlocking {
+      log.debug { "Goal created event for prisoner [$prisonNumber]" }
+      launch {
+        telemetryService.trackGoalCreateEvent(createdGoal)
+        timelineService.recordTimelineEvent(prisonNumber, goalCreatedTimelineEvent(createdGoal))
```

Comment:
I've decided against having a separate generic event, which in theory could have been used for either the Timeline or App Insights.

In reality, the events would have only existed in this class before being mapped to the `Timeline` domain (in order to pass to the `TimelineService`).